lms/djangoapps/grades/models.py

```diff
@@ -363,3 +365,76 @@ def _prepare_params_visible_blocks_id(cls, params):
         """
         params['visible_blocks_id'] = params['visible_blocks'].hash_value
         del params['visible_blocks']
+
+
+class PersistentCourseGrade(TimeStampedModel):
+    """
+    A django model tracking persistent course grades.
+    """
+
+    class Meta(object):
+        # Indices:
+        # (course_id, user_id) for individual grades
+        # (course_id) for instructors to see all course grades
+        # (user_id) for course dashboard
+        unique_together = [
+            ('course_id', 'user_id'),
+        ]
+
+    # primary key will need to be large for this table
+    id = UnsignedBigIntAutoField(primary_key=True)  # pylint: disable=invalid-name
+
+    # uniquely identify this particular grade object
+    user_id = models.IntegerField(blank=False, db_index=True)
+    course_id = CourseKeyField(blank=False, max_length=255)
+
+    # Information relating to the state of content when grade was calculated
+    subtree_edited_timestamp = models.DateTimeField('last content edit timestamp', blank=False)
+    course_version = models.CharField(u'Guid of latest course version', blank=True, max_length=255)
```

Comment:
This is not necessarily a GUID, by the way.  Different modulestores may choose to implement their course-versions differently.  Also, it's not the "latest course version".  Rather the version of the course at the time of computing the grade.  Perhaps the following phrase:  `Course content version identifier`?